src/core/mac/sub_mac.hpp

```diff
@@ -480,6 +480,17 @@ class SubMac : public InstanceLocator, private NonCopyable
     bool IsRadioFilterEnabled(void) const { return mRadioFilterEnabled; }
 #endif
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+    /**
+     * Configures wake-up listening parameters in all radios.
+     *
+     * @param[in]  aInterval  The WED listen interval in microseconds.
+     * @param[in]  aDuration  The WED listen duration in microseconds.
+     * @param[in]  aChannel   The wake-up channel.
+     */
+    void UpdateWakeupListening(uint32_t aInterval, uint16_t aDuration, uint8_t aChannel);
```

Comment:
The code here is using condition`aInterval == 0` to enable/disable the wake-up listening in `SubMac`. OT code is using the `mCslPeriod==0` to enable/disable CSL. But the sleepy-to-sleepy spec uses the `CslPeriod=0` to tear down an active link. So we have to update this kind of usage later. Could we add a function `SetWakeupListenEnabled(bool aEnabled)` to enable/ disable the wake-up listening to avoid the similar situation happens again?

Reply:
I'm not sure I understand the problem. When the WED receives a frame with both CST phase and period equal to zero, it understands that it should detach from WC, but this period won't be applied to `mCslPeriod`. See https://github.com/nrfconnect/sdk-openthread/commit/collab-thread-s2s#diff-b96f044b83202cb4c21ddc970d09d94de39b9ae425f4d02899deb7aa1fd279f4R2893-R2898

Could you please elaborate?

Reply:
The spec said that `A Wake-up Parent MAY implement eCSL`, so the WC may not use the eCSL to establish a link with WED. In this case, the WC can use the zero valued CSL-IE to terminate the link.

I think your new method is fine.

Reply:
I implemented updated method.

Reply:
> The spec said that `A Wake-up Parent MAY implement eCSL`, so the WC may not use the eCSL to establish a link with WED. In this case, the WC can use the zero valued CSL-IE to terminate the link.

I see, but the spec also says:

> In case the Wake-up Parent does not use eCSL, the communication link automatically falls back to regular CSL (in case the WED uses this) or to a regular SED/MED link (in case the WED operates as respectively a SED or MED while it is awake)

Which means a WED might not use regular CSL.

And also:

> A WED MUST support eCSL in order to support a Sleepy Parent that uses it.

Isn't then reasonable to support the CST=0 method only?

Reply:
> In case the Wake-up Parent does not use eCSL, the communication link automatically falls back to regular CSL (in case the WED uses this) or to a regular SED/MED link (in case the WED operates as respectively a SED or MED while it is awake)

My understanding is that, after the WP receives the Parent Request, if the WP supports the `eCSL`, the WP will include the `CST IE` and `CSL IE` in the Parent Response. The WP and WED use the `CST IE` and WP's `CSL IE` to establish the link with each other. When the WP wants to tear down the link, it just set the zero valued `CST IE`. If the WP only support the CSL, the WP will only include the `CSL IE` in the Parent Response, the WC and WED use each other's `CSL IE` to establish the link with each other. When the WP wants to tear down the link, it just set the zero valued `CSL IE`.

> A WED MUST support eCSL in order to support a Sleepy Parent that uses it.

The WED must support eCSL, but the WP may not use it.

Reply:
But what if the WED doesn't support CSL? Like a MED. Does it make sense for this kind of device to parse `CSL IE`?

Reply:
I don't know how to process this case, we may raise a discussion in Thread Group about this case.